### ISSUE TITLE
Hide / Show UI components

### ISSUE THREAD

**Issue by TobiasElemental** [OurMachinery/themachinery-public#1390]
Rects, texts, labels etc should have a hide/show feature in graph scripting.
Atm my hacky workaround is to set rects to scale 0,0,0,0 and text strings to "double space" in order to hide them during run-time.

**Comment by Tisten** [comment]
This should be possible to implement using this PR https://github.com/OurMachinery/themachinery/pull/969 right?

I only see this feature being used in unit tests where `enabled_disabled_opt_in` is set, so this might be a good initial use case for it @leonardo-ourmachinery?

**Comment by leonardo-ourmachinery** [comment]
Yes. This would be a perfect use case to the disabled/enabled components feature.

I tried to approach an initial implementation last week, but as I started I realized that we don't have a way to "talk" about components in the Entity Graph: so currently an hypothetical "Disable Component" Node would have to take something like a string hash as a parameter, and that is not ideal.

Once we have (and I think we should eventually have it) a way to "Pick" a `tm_component_type_t` in the Graph editor this will become super easy.

**Comment by Tisten** [comment]
I created nodes for this and I felt that it went rather smooth. But when extending the functionality I stumbled upon https://github.com/OurMachinery/themachinery-public/issues/1400

**Comment by Tisten** [comment]
I've now added this to our local build for UI text and rect components. Seems to work fine in my local tests. Will let Tobbe try it out now and can make complete PR when it is confirmed to work
